irohad/ametsuchi/query_executor_visitor.hpp

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_AMETSUCHI_QUERY_EXECUTOR_VISITOR_HPP
+#define IROHA_AMETSUCHI_QUERY_EXECUTOR_VISITOR_HPP
+
+#include <memory>
+
+#include <boost/variant/static_visitor.hpp>
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/permissions.hpp"
+
+namespace shared_model {
+  namespace interface {
+    class GetAccount;
+    class GetBlock;
+    class GetSignatories;
+    class GetAccountTransactions;
+    class GetAccountAssetTransactions;
+    class GetTransactions;
+    class GetAccountAssets;
+    class GetAccountDetail;
+    class GetRoles;
+    class GetRolePermissions;
+    class GetAssetInfo;
+    class GetPendingTransactions;
+    class GetPeers;
+    class Query;
+    class QueryResponse;
+  }  // namespace interface
+}  // namespace shared_model
+
+namespace iroha {
+  namespace ametsuchi {
+
+    using QueryExecutorResult =
+        std::unique_ptr<shared_model::interface::QueryResponse>;
+
+    class QueryExecutorVisitor
+        : public boost::static_visitor<QueryExecutorResult> {
+     public:
+      virtual ~QueryExecutorVisitor() = default;
+
+      QueryExecutorResult execute(const shared_model::interface::Query &qry);
+
+      virtual void setCreatorId(
+          const shared_model::interface::types::AccountIdType &creator_id) = 0;
```

Comment:
agreed offline that `execute(const shared_model::interface::Query &qry)` should suffice for interface type, and inheritance of `boost::static_visitor` should be removed too.